src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_INDEX = 10;
+    private static final double GROW_INDEX = 1.5;
+    private int maxSize = DEFAULT_INDEX;
+    private int sizeOfArray;
+    private T[] arrayDefault;
+    private T[] arrayTemporary;
+
+    public ArrayList() {
+        arrayDefault = (T[]) new Object[maxSize];
+    }
+
+    public void errorCheck(int index, int size) throws ArrayListIndexOutOfBoundsException {
```

Comment:
```suggestion
     private void rangeCheck(int index) {
```
no need to put size here - you have it in the class
move private methods after public